src/core/net/ip6.cpp

```diff
@@ -369,7 +385,7 @@ ThreadError Ip6::HandlePayload(Message &message, MessageInfo &messageInfo, uint8
 ThreadError Ip6::ProcessReceiveCallback(const Message &aMessage, const MessageInfo &messageInfo, uint8_t aIpProto)
 {
     ThreadError error = kThreadError_None;
-    Message *messageCopy = NULL;
+    Message *messageCopy;
```

Comment:
What is the advantage of doing this? Previously it makes sense since we had a check in exit label. IMHO messageCopy is used only when assigning to clone message, so we skip unnecessary one instruction to clear messageCopy pointer onthe stack. Saying that i am okay with changing this :)